common/stage.py

```diff
@@ -460,7 +479,188 @@ def seed_context( self, kll_file ):
 		'''
 		kll_file.context.initial_context( kll_file.lines, kll_file.data, kll_file )
 
-	def process( self ):
+	def process_connect_ids(self, kll_file, apply_offsets):
+		lines = kll_file.data.splitlines()
+
+		# Basic Tokens Spec
+		# TODO Storing these somewhere central might be a reasonable idea
```

Comment:
Is there some plans for storing the specs somewhere as there is a danger that they start diverging at some point and that can be a bit of a pain.

Reply:
Hmm. That might be a good idea. Would probably be called token.py, haha.

What's tricky is that the spec order matters a lot, so by putting them in a central place, you'd have to cross-reference back.

The other problem is that sometimes you have to tweak the regex in certain contexts because of a prior rule.
This is why I split up the tokenization and parsing (was really annoying to think about everything all the time).